packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/field/AdditionalPropertiesSchemaField.tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  FieldProps,
+  FormContextType,
+  RJSFSchema,
+  StrictRJSFSchema,
+} from '@rjsf/utils'
+import { isEqual } from 'lodash-es'
+import React, { useEffect, useState } from 'react'
+import FullWidthAlert from '../../FullWidthAlert/FullWidthAlert'
+import { convertToArray } from '../AnnotationEditorUtils'
+import { Grid, InputLabel } from '@mui/material'
+import { AdditionalPropertyContextProvider } from '../template/AdditionalPropertyContext'
+
+// Matches ####-##-##T##:##:##.###Z, e.g. 1970-01-01T12:00:000Z
+const ISO_TIMESTAMP_REGEX = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z/
+
+// Types that correspond to the different input fields that the annotation editor supports
+export enum PropertyType {
+  STRING = 'String',
+  INTEGER = 'Integer',
+  FLOAT = 'Float',
+  BOOLEAN = 'Boolean',
+  DATETIME = 'Datetime',
+}
+
+export function guessPropertyType(list: Array<unknown>): PropertyType {
+  if (list.length === 0) {
+    // The field was just added, so default to string
+    return PropertyType.STRING
+  } else if (
+    list.every(
+      item => typeof item === 'number' || item === 'NaN', // "NaN" is technically a float value
+    )
+  ) {
+    if (list.every(item => Number.isInteger(item))) {
+      return PropertyType.INTEGER
+    } else {
+      return PropertyType.FLOAT
+    }
+  } else if (list.every(item => typeof item === 'boolean')) {
+    return PropertyType.BOOLEAN
+  } else if (
+    list.every(item => typeof item === 'string') &&
+    (list as string[]).every((item: string) => !!ISO_TIMESTAMP_REGEX.exec(item))
+  ) {
+    return PropertyType.DATETIME
+  }
+  // otherwise, default type is 'string'
+  return PropertyType.STRING
+}
+
+export function transformDataFromPropertyType(
+  list: Array<any>,
+  propertyType: PropertyType,
+) {
+  switch (propertyType) {
+    case PropertyType.INTEGER:
+      return list.map(item =>
+        Number.isNaN(Number(item)) ? 0 : Math.floor(Number(item)),
+      )
+
+    case PropertyType.FLOAT:
+      return list.map(item => {
+        const asFloat = parseFloat(item as string)
+        if (Number.isNaN(asFloat)) {
+          return 'NaN'
+        } else if (Number.isInteger(asFloat)) {
+          return asFloat.toFixed(1)
+        } else {
+          return asFloat
+        }
+      })
+    case PropertyType.DATETIME:
+      return list.map(item => {
+        if (typeof item === 'string' && ISO_TIMESTAMP_REGEX.exec(item)) {
+          return item
+        } else {
+          return new Date().toISOString()
+        }
+      })
+    case PropertyType.BOOLEAN:
+      return list.map(item => !!item)
+    case PropertyType.STRING:
+    default:
+      return list.map(item => String(item))
+  }
+}
+
+/**
+ * Maps a Synapse Annotation PropertyType to a JSON Schema that captures the type and format.
+ * @param propertyType
+ */
+export function getSchemaForPropertyType(propertyType: PropertyType) {
```

Comment:
Previously mapped `PropertyType` to a `Widget`. Now, map `PropertyType` to a schema and RJSF will render the correct widget, since it's already doing that for the rest of the form.